src/java/main/org/apache/zookeeper/common/ZKConfig.java

```diff
@@ -107,14 +99,33 @@ private void init() {
      * this configuration.
      */
     protected void handleBackwardCompatibility() {
-        properties.put(SSL_KEYSTORE_LOCATION, System.getProperty(SSL_KEYSTORE_LOCATION));
-        properties.put(SSL_KEYSTORE_PASSWD, System.getProperty(SSL_KEYSTORE_PASSWD));
-        properties.put(SSL_TRUSTSTORE_LOCATION, System.getProperty(SSL_TRUSTSTORE_LOCATION));
-        properties.put(SSL_TRUSTSTORE_PASSWD, System.getProperty(SSL_TRUSTSTORE_PASSWD));
-        properties.put(SSL_AUTHPROVIDER, System.getProperty(SSL_AUTHPROVIDER));
         properties.put(JUTE_MAXBUFFER, System.getProperty(JUTE_MAXBUFFER));
         properties.put(KINIT_COMMAND, System.getProperty(KINIT_COMMAND));
         properties.put(JGSS_NATIVE, System.getProperty(JGSS_NATIVE));
+
+        ClientX509Util clientX509Util = new ClientX509Util();
```

Comment:
should we have client stuff in this diff? we are just doing quorum security. right?

Reply:
The naming confused me a bit too, but yeah, this is quorum security only.

Reply:
I'm thinking of adding a short comment to these classes for clarification.

Reply:
@anmolnar good idea. Or maybe rename the class? Something like `QuorumClientX509Util`?

Reply:
Not sure that would be better either. Looks like `ClientCnxnSocketNetty` uses it too to create SSLContext.

Reply:
Hmm, I don't know if we necessarily want to use the same SSL settings for that, but maybe we can figure it out later. Is `ClientCnxnSocketNetty` used for connections between ensemble servers and ZK clients?

Reply:
That's correct. So it basically turned out that `ClientX509Util` is somewhat related to client communication too. Later, in a separate ticket, we can either pull everything together or make the separation more strict, but I wouldn't touch it for now.